utils/zerovec/src/ule/mod.rs

```diff
@@ -33,8 +38,30 @@ where
     /// Implementations of this method may involve `unsafe{}` blocks to cast the pointer to the
     /// correct type. It is up to the implementation to reason about the safety. Keep in mind that
     /// `&[Self]` and `&[u8]` may have different lengths.
+    ///
+    /// Ideally, implementations call [`ULE::from_byte_slice_unchecked()`] after validation.
     fn parse_byte_slice(bytes: &[u8]) -> Result<&[Self], Self::Error>;
 
+    /// Takes a byte slice, `&[u8]`, and return it as `&[Self]` with the same lifetime, assuming that
+    /// this byte slice has previously been run through [`ULE::parse_byte_slice()`] with success.
+    ///
+    /// There is no need to perform any validation here, this should almost always be a straight pointer
+    /// cast.
+    ///
+    /// # Safety
+    ///
+    /// ## Callers
+    /// Callers of this method must take care to ensure that `bytes` was previously passed through
+    /// [`ULE::parse_byte_slice()`] with success (and was not changed since then).
+    ///
+    /// ## Implementors
+    /// This method _must_ be implemented to return the same result as [`ULE::parse_byte_slice()`].
+    ///
+    /// Implementations of this method may involve `unsafe{}` blocks to cast the pointer to the
+    /// correct type. It is up to the implementation to reason about the safety, assuming the invariant
+    /// above.
+    unsafe fn from_byte_slice_unchecked(bytes: &[u8]) -> &[Self];
```

Comment:
Why do you need to add the new trait method?  It seems orthogonal to the problem at hand, which is enabling `VZV<Vec<T>>`.

Reply:
(VarULE and ULE have two different from_byte_slice_unchecked functions)